### ISSUE TITLE
Settings option to disable overtype mode.

### ISSUE THREAD

**Issue by ogrotten** [microsoft/vscode#238668]
Type: <b>Feature Request</b>

I don't begrudge people that use overtype/overstrike mode. I expected that there would be a settings entry to control allowing the previous behaviour or the new behaviour. This seems to usually be the case for new features that change an existing behaviour. 

Accessibility wise, if one's keyboard aim for Home or Del keys isn't perfect, a user can end up in overstrike mode without realizing. The solution for such bad aim, combined with no-choice overstrike mode, is to remove the insert key to eliminate accidents. 

Please implement a settings option that allows a user to "disable overtype". 

VS Code version: Code 1.96.3 (91fbdddc47bc9c09064bf7acf133d22631cbf083, 2025-01-09T18:14:09.060Z)
OS version: Windows_NT x64 10.0.22631
Modes:
Remote OS version: Linux x64 5.15.153.1-microsoft-standard-WSL2

**Comment by meganrogge** [comment]
I'm not sure what you mean by overtype mode, could you pls clarify?

**Comment by ogrotten** [comment]
> I'm not sure what you mean by overtype mode, could you pls clarify?

https://code.visualstudio.com/updates/v1_96#_overtype-mode

This feature has been added as always available. I'm suggesting that there be a toggle to disable the feature. 

Accessibility wise, due to my own physical limitations, my aim to keys near the INSERT key isn't great. Which means I accidentally hit that key and turn on "overtype mode" without wanting to. Without a toggle, I have removed the INSERT key to prevent accidental hits. 

The feature request is to add a settings toggle to disable "overtype mode" entirely, so that pressing INSERT does nothing. . . like VSCode was before the addition of the feature.

**Comment by meganrogge** [comment]
Thank you for clarifying. Adding @aiday-mar to this, as she is the one who created this mode.

**Comment by aiday-mar** [comment]
Hi thank you for filing this issue, so there is a way to unbind keys to commands by adding the following to your keyboard shortcuts json file:

{ "key": "ins", "command": "-editor.action.toggleOvertypeInsertMode" }

I believe the code should be `ins`, but I am not quite sure. You can also right click on the keyboard shortcut for `editor.action.toggleOvertypeInsertMode` in the keyboard shortcuts menu and select the option `Remove Keybinding`. Let me know if this solution works for you, or if it is not enough.

**Comment by ogrotten** [comment]
> You can also right click on the keyboard shortcut for `editor.action.toggleOvertypeInsertMode` in the keyboard shortcuts menu and select the option `Remove Keybinding`.

This did the trick thank you very much for the suggestion.

To be clear, un-assigning the shortcut key for the feature has the effect of making the feature unavailable. 

I believe this issue can be closed.

**Comment by aiday-mar** [comment]
Great thanks

**Comment by aiday-mar** [comment]
That's a good point, could you file a new issue for this and we can see how the community reacts to this issue?
